cclient/serialization/json/tests/get_neighbors.c

```diff
@@ -26,28 +26,45 @@ void test_get_neighbors_serialize_response(void) {
   serializer_t serializer;
   init_json_serializer(&serializer);
   char_buffer_t* out = char_buffer_new();
+
   char const * TEST_JSON_TEXT =
       "{\"neighbors\":[{"
-      "\"address\":\"" TEST_NEIGHBORS_NEIGHBOR1
-      "\",\"numberOfAllTransactions\":" STR(TEST_NEIGHBORS_NUMALLTX1)
-      ",\"numberOfInvalidTransactions\":" STR(TEST_NEIGHBORS_NUMINVALIDTX1)
-      ",\"numberOfNewTransactions\":" STR(TEST_NEIGHBORS_NUMNEWTX1)
-      "},{\"address\":\"" TEST_NEIGHBORS_NEIGHBOR2
-      "\",\"numberOfAllTransactions\":" STR(TEST_NEIGHBORS_NUMALLTX2)
-      ",\"numberOfInvalidTransactions\":" STR(TEST_NEIGHBORS_NUMINVALIDTX2)
-      ",\"numberOfNewTransactions\":" STR(TEST_NEIGHBORS_NUMNEWTX2) "}]}";
+      "\"address\":\"" TEST_NEIGHBORS_NEIGHBOR1 "\""
+      ",\"numberOfAllTransactions\":" STR(TEST_NEIGHBORS_NUMALLTX1)
+      ",\"numberOfRandomTransactionRequests\":" STR(TEST_NEIGHBORS_NUMRANDTX1)
+    ",\"numberOfNewTransactions\":" STR(TEST_NEIGHBORS_NUMNEWTX1)
+    ",\"numberOfInvalidTransactions\":" STR(TEST_NEIGHBORS_NUMINVALIDTX1)
+    ",\"numberOfStaleTransactions\":" STR(TEST_NEIGHBORS_NUMSTALETX1)
+    ",\"numberOfSentTransactions\":" STR(TEST_NEIGHBORS_NUMSENTTX1)
+    ",\"connectiontype\":\"" TEST_NEIGHBORS_CONNNECTION1 "\"},{"
+    "\"address\":\"" TEST_NEIGHBORS_NEIGHBOR2 "\""
+      ",\"numberOfAllTransactions\":" STR(TEST_NEIGHBORS_NUMALLTX2)
+      ",\"numberOfRandomTransactionRequests\":" STR(TEST_NEIGHBORS_NUMRANDTX2)
+    ",\"numberOfNewTransactions\":" STR(TEST_NEIGHBORS_NUMNEWTX2)
+    ",\"numberOfInvalidTransactions\":" STR(TEST_NEIGHBORS_NUMINVALIDTX2)
+    ",\"numberOfStaleTransactions\":" STR(TEST_NEIGHBORS_NUMSTALETX2)
+    ",\"numberOfSentTransactions\":" STR(TEST_NEIGHBORS_NUMSENTTX2)
+    ",\"connectiontype\":\"" TEST_NEIGHBORS_CONNNECTION2"\"}]}";
 
   get_neighbors_res_t* res = get_neighbors_res_new();
-  char_buffer_t* test_neighbors_neighbor1 = char_buffer_new();
-
-  char_buffer_set(test_neighbors_neighbor1, TEST_NEIGHBORS_NEIGHBOR1);
-  char_buffer_t* test_neighbors_neighbor2 = char_buffer_new();
-  char_buffer_set(test_neighbors_neighbor2, TEST_NEIGHBORS_NEIGHBOR2);
-  ret = get_neighbors_res_add_neighbor(res, test_neighbors_neighbor1, TEST_NEIGHBORS_NUMALLTX1,
-                                       TEST_NEIGHBORS_NUMINVALIDTX1, TEST_NEIGHBORS_NUMNEWTX1);
+  char_buffer_t* test_neighbors_neighbor_add1 = char_buffer_new();
```

Comment:
memory leak.
see https://github.com/iotaledger/entangled/pull/1053#issuecomment-479373814
change `char_buffer`  type to `char const * const`, allocate/free memory inside the get_neighbors object.